homeassistant/components/nanoleaf/device_trigger.py

```diff
@@ -52,6 +61,44 @@ async def async_get_triggers(
     ]
 
 
+async def async_get_trigger_capabilities(
+    hass: HomeAssistant, config: ConfigType
+) -> dict[str, vol.Schema]:
+    """List trigger capabilities."""
+    device_registry = dr.async_get(hass)
+    device_entry = device_registry.async_get(config[CONF_DEVICE_ID])
+    if device_entry is None:
+        raise DeviceNotFound(f"Device ID {config[CONF_DEVICE_ID]} is not valid")
+    if config[CONF_TYPE] in TRIGGER_TYPES_THAT_REPORT_PANEL_ID:
+        if nanoleaf := get_nanoleaf_connection_by_device_id(
+            hass, config[CONF_DEVICE_ID]
+        ):
+            if panels := nanoleaf.panels:
```

Comment:
If I can cache the panels during (1) device assessment and (2) nanoleaf layout update events, I think we can get rid of this entirely? It would also remove the need for `get_nanoleaf_connection_by_device_id`.

Reply:
(aionanoleaf caches this)